### PR TITLE
Fixed interface with z3

### PR DESCRIPTION
**Pull request by ferresb**
There is a problem with the interface with z3, in `src/z3overlay.ml` (line 233): 
`let inter reL = Seq.mk_re_intersect ctx (List.length reL) reL` should be `let inter reL = Seq.mk_re_intersect ctx reL` with the current API.

I also modified the tests in consequence.

### CODE REVIEW

File: test/simple.ml [hunk 1]
```diff
@@ -15,6 +16,7 @@ let () =
   (* We create new SMT variables and specify their types. *)
   let x = Symbol.declare Real "x" in
   let y = Symbol.declare Real "y" in
+  let z = Symbol.declare Bool "y" in
```
Comment (Drup) [inline test/simple.ml]: Shouldn't it be named `z` ?

File: test/simple.ml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 (* We hide the context by instanciating a functor: *)
-module Z = ZZ3.Make (struct let ctx = Z3.mk_context [] end)
+module Z = Z3overlay.Make (struct let ctx = Z3.mk_context [] end)
+(* module Z = ZZ3.Make (struct let ctx = Z3.mk_context [] end) *)
```
Comment (Drup) [inline test/simple.ml]: Get rid of the old one, it's deprecated !

File: test/simple.ml (hunk 1 above)
Comment (ferresb) [reply test/simple.ml]: It should not even exist, it's obviously a mistake.

File: test/simple.ml (hunk 2 above)
Comment (ferresb) [reply test/simple.ml]: Indeed.

**Pull request merged**